### DISCUSSION TITLE
v1.12.3 (Beta)

### DISCUSSION THREAD

**Discussion by deividAlfa** [deividAlfa/stm32_soldering_iron_controller#91] [Announcements]
- Detect flash size at boot, hopefully this will be the fix for some controllers.
  I suspect some controllers are using High density STM32 (RC/RD/RE), which have a larger page size.
  If so, this should fix the big issue once for all.

- Fix a bug in Tip Editor not selecting the tip choosen in the Tip Editor screen.

     Please report any bugs found!

**Comment by iopia** [comment]
Good afternoon! on version 1.12.3 an error appears after changing the menu "bad checksum error/ flash error"
KSGER_v2_OLED.zip

**Comment by iopia** [comment]
![IMG_20240103_114346](https://github.com/deividAlfa/stm32_soldering_iron_controller/assets/8014237/7bc1d7e5-09d6-4b49-bd46-15e8138bd08a)
![IMG_20240103_114354](https://github.com/deividAlfa/stm32_soldering_iron_controller/assets/8014237/b522d77a-c17f-4d09-a836-20036a21a380)
![IMG_20240103_114422](https://github.com/deividAlfa/stm32_soldering_iron_controller/assets/8014237/b57b8c29-c5ac-4892-98aa-5d8312fd7b6c)

**Reply by deividAlfa** [reply]
Does it work gain after the bad checksum?
Happens only with SYSTEM settings, or also with PROFILE or ADDONS?
Please upload picture of the stm32.
Thanks!

**Reply by iopia** [reply]
Does it work gain after the bad checksum?- no, the error appears after reboot
Happens only with SYSTEM settings, or also with PROFILE or ADDONS?- everywhere

there is no error on version 1.12.2

stm32f103cbu6
![WhatsApp Image 2024-01-03 at 14 14 47_5d0e6362](https://github.com/deividAlfa/stm32_soldering_iron_controller/assets/8014237/50e09d26-0ae8-4f70-a994-62f4a57cea73)

**Reply by deividAlfa** [reply]
I mean, works again after the checksum error or it keeps showing the checksum error everytime?

**Reply by iopia** [reply]
it cannot recover the checksum and always shows an error.

I install the firmware with complete erasure.

**Comment by deividAlfa** [comment]
Mine is also a CB, but works perfectly, I suspect this is a relabeled clone...
Please run this tests and reprt the results:
https://www.eevblog.com/forum/reviews/stm32-oled-digital-soldering-station-for-t12-handle/msg5253981/#msg5253981

**Comment by sst311212** [comment]
It should be getSystemSettings().
https://github.com/deividAlfa/stm32_soldering_iron_controller/blob/02224b6965ad9b0bb85bed3e0757f3ebebe579d8/Core/Src/user_main.c#L134

It should be backupTemp?
https://github.com/deividAlfa/stm32_soldering_iron_controller/blob/02224b6965ad9b0bb85bed3e0757f3ebebe579d8/Drivers/graphics/gui/screens/calibration_screen.c#L212

**Reply by deividAlfa** [reply]
Yep, will fix it later, thanks.

**Reply by sst311212** [reply]
It breaks selection in combo menu again, currentScroll is not behave something you want.
https://github.com/deividAlfa/stm32_soldering_iron_controller/blob/02224b6965ad9b0bb85bed3e0757f3ebebe579d8/Drivers/graphics/gui/screens/tip_list_screen.c#L21

**Reply by deividAlfa** [reply]
Oops, fixed. Thanks

**Comment by deividAlfa** [comment]
Locked as [1.12.4-beta](https://github.com/deividAlfa/stm32_soldering_iron_controller/discussions/94) is out.
